### PR TITLE
AM-2994 allow making email optional for new users created via management api

### PR DESCRIPTION
**Pull request by mregulski-gravitee**
## :id: Reference related issue. 
AM-2994

## :pencil2: A description of the changes proposed in the pull request
Introduces a new config property in `gravitee.yml`: `user.email.required: (boolean)`, set to `true` by default.
If `false`, management API will allow creating users without email addresses. 

The new property is exposed on a new config endpoint: `configuration/users/emailRequired`, allowing UI to adapt appropriately.

### CODE REVIEW

File: gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/platform/configuration/ConfigurationResource.java [hunk 1]
```diff
@@ -85,4 +86,12 @@ public void getSpelGrammar(@Suspended final AsyncResponse response) {
             .subscribe(response::resume, response::resume);
     }
 
+    @GET
+    @Path("users/emailRequired")
```
Comment (mstasgravitee) [inline gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/platform/configuration/ConfigurationResource.java:90]: Path has to be lowercase: "users/email-required"

Shouldn't we think about more generic configuration endpoint for easy adaptability in the future? What do you think?

Comment (mregulski-gravitee) [reply gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/platform/configuration/ConfigurationResource.java:90]: I agree on both points.
The more general endpoint, might be a good idea (we could then just load it once in `main.ts`, together with `constants.json` and `build.json`), but I don't think we currently have enough configuration that needs such an endpoint to know how to best structure it.
I think for now it's good enough that these few config elements are grouped under a single resource.

File: gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/platform/configuration/ConfigurationResource.java [hunk 2]
```diff
@@ -85,4 +86,12 @@ public void getSpelGrammar(@Suspended final AsyncResponse response) {
             .subscribe(response::resume, response::resume);
     }
 
+    @GET
+    @Path("users/email-required")
+    @Produces(jakarta.ws.rs.core.MediaType.APPLICATION_JSON)
```
Comment (mstasgravitee) [inline gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/platform/configuration/ConfigurationResource.java:91]: produces JSON in annotation but it is returned boolean

File: gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/platform/configuration/ConfigurationResource.java [hunk 3]
```diff
@@ -79,10 +83,19 @@ public void getAlertServiceStatus(@Suspended final AsyncResponse response) {
...
+                .subscribe(response::resume, response::resume);
+    }
+
+    @GET
+    @Path("users/email-required")
+    @Produces(jakarta.ws.rs.core.MediaType.APPLICATION_JSON)
+    public void getUserEmailRequired(@Suspended final AsyncResponse response) {
+        var emailRequired = environment.getProperty(UserEmail.PROPERTY_USER_EMAIL_REQUIRED, boolean.class, true);
+        Single.just(Map.of("emailRequired", emailRequired))
+                .subscribe(response::resume, response::resume);
```
Comment (mstasgravitee) [inline gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/platform/configuration/ConfigurationResource.java:98]:
```suggestion
        response.resume(Map.of("emailRequired", emailRequired));
```

**Review by mstasgravitee — APPROVED** [review]

**Pull request merged**